service/src/main/java/bio/terra/workspace/service/workspace/flight/WorkspaceDeleteFlight.java

```diff
@@ -73,35 +71,53 @@ public WorkspaceDeleteFlight(FlightMap inputParameters, Object applicationContex
         new DeleteGcpProjectStep(
             appContext.getCrlService(), appContext.getGcpCloudContextService()),
         cloudRetryRule);
+
+    addStep(
+        new EnsureNoWorkspaceChildrenStep(appContext.getSamService(), userRequest, workspaceUuid));
+
     addStep(
         new DeleteAzureContextStep(appContext.getAzureCloudContextService(), workspaceUuid),
         cloudRetryRule);
     addStep(
         new DeleteAwsContextStep(appContext.getAwsCloudContextService(), workspaceUuid),
         cloudRetryRule);
+    addAuthZSteps(appContext, inputParameters, userRequest, workspaceUuid, terraRetryRule);
+    addStep(
+        new DeleteWorkspaceStateStep(appContext.getWorkspaceDao(), workspaceUuid), terraRetryRule);
+  }
+
+  /**
+   * Adds steps to delete sam resources owned by WSM Workspace authz is handled differently
```

Comment:
Nit: I think you are missing a period/semi-colon after "WSM Workspace".

Reply:
I'm missing a line break
I moved the inline comment from the block of code this was factoring out to the top of the method, and I think some formatting got weird.

Reply:
It was actually that without a period, running spotlessApply tries to reformat it.